cf-agent/files_editline.c

```diff
@@ -1417,7 +1417,7 @@ static bool SanityCheckDeletions(const Attributes *a, const Promise *pp)
         {
             Log(LOG_LEVEL_ERR,
                 "Makes no sense to promise multi-line delete with not_matching. Cannot be satisfied for all lines as a block.");
-            // FIXME: This function always returns true (!)
+            return false;
```

Comment:
Are you sure this doesn't change any behavior?

Reply:
The return value of the function is never evaluated, so it doesn't. It might be better to just set it to `void`, but shouldn't the "SanityCheck" return failure if it also gives an error? Who knows.

Reply:
@karlhto That's not true:

https://github.com/cfengine/core/blob/6403c3eb9784db75f87abcf99693e144c1d1a9b8/cf-agent/files_editline.c#L360

Reply:
I agree with the change though, but it will/might change behavior. So I would probably do it in a separate PR, and try to do some extra (manual) testing.

Reply:
Whoops, think I thought about its "Insertions" variant instead. Thanks for noticing. :^)